### ISSUE TITLE
Refactor initialization of web UI

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#3086]
Our current initialization code is messy and fragile with a bunch of synchronous code that blocks the UI and must be executed in a particular well defined order or things break. #3084 is an example of a recent symptom of this, but we've seen a number of related issues.

### Proposed solution

Any [place](https://github.com/OpenRefine/OpenRefine/search?q=async%3A+false&type=Code) that uses `async: false` in an AJAX call is suspicious and any initialization of non-visible elements should be deferred. Any required initializations should be done in parallel to the extent possible with only the necessary dependencies blocking (e.g. language loading and other initializations which don't depend on translations should happen in parallel.

Promises are one technology which could potentially be used to achieve this cleanly.

Not doing the initializations at startup also implies moving the necessary code to be done, if necessary, before a requested tab/screen/dialog gets made visible.

As an example, the Google Data importer is doing all of this during startup:
```
Refine.GDataSourceUI._listDocuments 
Refine.GDataSourceUI.attachUI
Refine.CreateProjectUI.addSourceSelectionUI 
Refine.GDataImportingController 
Refine.CreateProjectUI._initializeUI
```

This needs to be pared back significantly to just registering the necessary hooks to be invoked later.
